### ISSUE TITLE
class.sources that are added in a `forLinux` clause are not built

### ISSUE THREAD

**Issue by umlaeute** [pure-data/pd-lib-builder#61]
i just stumbled upon this while modernizing [hcs](https://github.com/pd-externals/hcs).

if i have an objectclass that should only build for a given platform, the documentation says to wrap it into a `define forXXX` clause:
https://github.com/pure-data/pd-lib-builder/blob/f087faf44112350cbc4e3809c51ef55aad2c68cd/Makefile.pdlibbuilder#L158-L163

However, doing so does not actually work: the specified classes are not build at all.
However, the `define` *does work*, as can be seen from other variables used in the clause.

consider the following replacement of the [helloworld](https://github.com/pure-data/helloworld) Makefile:

~~~make
# Makefile to build class 'helloworld' for Pure Data.
# Needs Makefile.pdlibbuilder as helper makefile for platform-dependent build
# settings and rules.

# library name
lib.name = helloworld

# input source file (class name == source file basename)
class.sources =
datafiles = helloworld-meta.pd README.md

define forLinux
  class.sources += test.c
  datafiles += helloworld-help.pd
endef

# include Makefile.pdlibbuilder from submodule directory 'pd-lib-builder'
PDLIBBUILDER_DIR=pd-lib-builder/
include $(PDLIBBUILDER_DIR)/Makefile.pdlibbuilder
~~~

Running `make` does not do anything.
Running `make install` will install the following files:

~~~sh
$ ls -1 .../helloworld/
helloworld-help.pd
helloworld-meta.pd
README.md
~~~

so the "helloworld.pd_linux" is missing, but the "helloworld-help.pd" gets installed.

**Comment by umlaeute** [comment]
it seems this never worked for any released version of pd-lib-builder.

i'm testing like this:

~~~sh
(cd pd-lib-builder; git tag | egrep "^v") | while read tag
do
    rm *.o *_*
    (cd pd-lib-builder/; git checkout ${tag})
    make && make install DESTDIR=$(pwd)/${tag}
    rm *.o *_*
done

find v*/ -type f  -name "*helloworld*" -not -name "*.pd"
~~~

**Comment by katjav** [comment]
This can't work indeed. The `define forXXX` method was added in response to issue #7 (Enable setting flags for OS). It works for flags but the example `class.sources += linuxthing.c` must have been made up without testing. Running make vars shows that cumulative variables like `classes.sources` and `classes.objects` are empty. The reason is because these variables are expanded before the platform detection section.

Now I tried swapping these sections and in that case the cumulative variables for file names are expanded correctly, and it does build. Actually I like this order of sections better; platform detection first, then build flags accumulation, then file name accumulation. Also the platform-dependent name definition of  `shared.lib` as discusssed in #59 would get a more logical place. But variable reshuffling in a makefile is hazardous and can easily give regressions, so I'd like to give this some more thought and testing.
